inference-engine/src/inference_engine/include/ie/ie_blob.h

```diff
@@ -239,6 +261,17 @@ class INFERENCE_ENGINE_API_CLASS(Blob) {
         return std::accumulate(std::begin(dims), std::end(dims), (size_t)1, std::multiplies<size_t>());
     }
 
+    /**
+     * @deprecated Cast to MemoryBlob and use its API instead.
+     * @brief Multiplies the dimension vector values. Size of a scalar is 1 instead of 0 as for product.
+     *
+     * @param dims Reference to a vector with dimension values of type size_t
+     * @return Result of multiplication
+     */
+    static size_t properProduct(const SizeVector& dims) noexcept {
+        return std::accumulate(std::begin(dims), std::end(dims), (size_t)1, std::multiplies<size_t>());
+    }
```

Comment:
can it be a lambda inside the `setShape`?

Reply:
Lower and upper bounds are hints. It is aux information that can be completely ignored by a plugin if it can work with generally dynamic dimensions w/o bounds.